packages/datetime/src/dateRangePicker.tsx

```diff
@@ -56,6 +57,13 @@ export interface IDateRangePickerProps extends IDatePickerBaseProps, IProps {
     onChange?: (selectedDates: DateRange) => void;
 
     /**
+     * Called when the user changes the hovered date range, either from mouseenter or mouseleave.
+     * When triggered from mouseenter, it will pass the date range that would result from next click.
+     * When triggered from mouseleave, it will pass `[null, null]`.
```

Comment:
@leebyp After building the whole `DateRangeInput` prototype on another branch, I can say this approach works empirically despite its clunkiness. That said, I'd love to figure out a way around this. The problem is that `mouseleave` on the old hovered date triggers before `mouseenter` on the new hovered date—maybe we can cancel the invocation of `this.props.onHoverChange` in `handleDayMouseLeave` if a `mouseenter` happens within the next `X` milliseconds? That feels kinda gross though.